client/helpers.go

```diff
@@ -79,38 +80,34 @@ func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
 		if err != nil {
 			return err
 		}
-		r, _, err := repo.GetDelegation(c.Scope())
-		if _, ok := err.(data.ErrNoSuchRole); err != nil && !ok {
-			// error that wasn't ErrNoSuchRole
+		_, err = repo.GetDelegationRole(c.Scope())
+		if _, ok := err.(data.ErrInvalidRole); err != nil && !ok {
+			// error that wasn't ErrInvalidRole
 			return err
 		}
+
 		if err == nil {
-			// role existed, attempt to merge paths and keys
-			if err := r.AddPaths(td.AddPaths); err != nil {
-				return err
-			}
-			return repo.UpdateDelegations(r, td.AddKeys)
-		}
-		// create brand new role
-		r, err = td.ToNewRole(c.Scope())
-		if err != nil {
-			return err
+			// role existed, merge paths and keys to this copy of the delegation role
+			addTD := changelist.TufDelegation{AddPaths: td.AddPaths, AddKeys: td.AddKeys}
+			return repo.UpdateDelegations(c.Scope(), addTD)
```

Comment:
The more I look at it the more I dislike passing the change object to the repo. I feel like we're setting ourselves up for creating circular imports we'll have to fix down the road. It might make sense, as verbose as it is, to pass the constituent parts of the TufDelegation object as individual args to UpdateDelegations.